### PR TITLE
fixed addon lint issues

### PR DESCRIPTION
**Pull request by dannycoates**
follow up to #2120

### CODE REVIEW

File: addon/src/lib/middleware/Hub.js [hunk 1]
```diff
@@ -14,8 +14,8 @@ export default class Hub {
   dispatch: Dispatch;
   ports: Set<EventEmitter>;
   constructor() {
-    this.dispatch = () =>
-      console.error('Hub cannot use dispatch() before middleware()');
+    this.dispatch = () => // eslint-disable-next-line no-console
```
Comment (pdehaan) [inline addon/src/lib/middleware/Hub.js:17]: Had to read this a couple times... looks like an arrow function to nowhere, but I assume it in fact references the `console.error` on the following line (which is now unindented below. Not sure if we want to throw some free `{`..`}` around the arrow function to make it clearer (or not, whatever).

But it passes `npm run lint` and still seems to lint after running `npm run format`.

File: addon/test/test-migrations.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* global describe beforeEach it */
+/* global describe it */
```
Comment (pdehaan) [inline addon/test/test-migrations.js:1]: If you don't want to manage these `global` variables per-file, we could also set an additional `env` of `mocha: true` in the addon/.eslintrc.yml file (or even just add extra env for tests in addon/test/.eslintrc.yml).

File: addon/src/lib/middleware/Hub.js [hunk 3]
```diff
@@ -14,8 +14,8 @@ export default class Hub {
...
+    console.error('Hub cannot use dispatch() before middleware()');
```
Comment (pdehaan) [inline addon/src/lib/middleware/Hub.js:18]: Offtopic: Should we do something more than `console.error()` here, like emit some error to Sentry or whatever?
Or do we care if errors get sent into a Browser Console abyss?

**Review by pdehaan — APPROVED** [review]
R+ with a few nit comments which can all be safely mocked and ignored. Merge at your convenience.

**Comment by pdehaan** [comment]
Also, a few undoubtedly unrelated warnings buried in the Circle-CI logs:

```
./bin/circleci/test-frontend.sh
+ npm run lint

> testpilot@1.0.0 lint /home/ubuntu/testpilot
> gulp scripts-lint styles-lint

[22:18:56] Using gulpfile ~/testpilot/gulpfile.js
[22:18:56] Starting 'scripts-lint'...
[22:18:56] Starting 'styles-lint'...
[22:19:00] Finished 'styles-lint' after 4.14 s
[22:19:01] Finished 'scripts-lint' after 4.98 s
+ MOCHA_FILE=/tmp/circle-junit.x889vdj/junit/test-results.xml
+ npm run test:ci

> testpilot@1.0.0 test:ci /home/ubuntu/testpilot
> NODE_ENV=test nyc mocha --recursive --reporter mocha-junit-reporter --require frontend/test-setup.js frontend/test

Warning: Failed prop type: The prop `locale` is marked as required in `NewsletterForm`, but its value is `undefined`.
    in NewsletterForm
Warning: Failed prop type: The prop `locale` is marked as required in `NewsletterFooter`, but its value is `undefined`.
    in NewsletterFooter
Warning: Failed form propType: You provided a `checked` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultChecked`. Otherwise, set either `onChange` or `readOnly`. Check the render method of `NewsletterForm`.
Warning: Failed prop type: The prop `locale` is marked as required in `View`, but its value is `undefined`.
    in View
Warning: Each child in an array or iterator should have a unique "key" prop. Check the top-level render call using <ul>. See https://fb.me/react-warning-keys for more information.
    in li
Warning: Failed prop type: The prop `subscribeToBasket` is marked as required in `EmailDialog`, but its value is `undefined`.
    in EmailDialog
Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `ExperimentDetail`. See https://fb.me/react-warning-keys for more information.
    in li (created by ExperimentDetail)
    in ExperimentDetail (created by Unknown)
    in Unknown

=============================== Coverage summary ===============================
Statements   : 96.46% ( 1797/1863 )
Branches     : 86.73% ( 340/392 )
Functions    : 94.41% ( 591/626 )
Lines        : 96.65% ( 1733/1793 )
================================================================================
```

Not sure if any of those are important, but I'll shut up now since this should be a separate bug for easier tracking.

---

**UPDATED:** Filed as #2122

File: addon/src/lib/middleware/Hub.js (hunk 1 above)
Comment (dannycoates) [reply addon/src/lib/middleware/Hub.js:17]: yeah, that got formatted weirdly by `prettier`

File: addon/src/lib/middleware/Hub.js (hunk 3 above)
Comment (dannycoates) [reply addon/src/lib/middleware/Hub.js:18]: This falls into the "should never happen" bin. I'll even tempt fate and say it'd never make it past QA. So its just a hint to me if I grossly bork it up.

**Pull request merged**